kinisi/parser.py

```diff
@@ -147,19 +147,33 @@ class PymatgenParser(Parser):
     :param progress: Whether to show a progress bar when reading in the structures. Optional, defaults to `True`.
     """
 
-    def __init__(self,
-                 structures: List['pymatgen.core.structure.Structure'],
-                 specie: Union['pymatgen.core.periodic_table.Element', 'pymatgen.core.periodic_table.Specie'],
-                 time_step: sc.Variable,
-                 step_skip: sc.Variable,
-                 dt: sc.Variable = None,
-                 dimension: str = 'xyz',
-                 distance_unit: sc.Unit = sc.units.angstrom,
-                 progress: bool = True):
+    def __init__(
+        self,
+        structures: List['pymatgen.core.structure.Structure'],
+        specie: Union['pymatgen.core.periodic_table.Element', 'pymatgen.core.periodic_table.Specie'],
+        time_step: sc.Variable,
+        step_skip: sc.Variable,
+        dt: sc.Variable = None,
+        dimension: str = 'xyz',
+        distance_unit: sc.Unit = sc.units.angstrom,
+        progress: bool = True,
+        specie_indices: sc.Variable = None,
```

Comment:
I think that `progress` should be the final keyword argument. Also, can we add something to the docstring for the new arguments?